internal/ui/list/list.go

```diff
@@ -631,6 +709,49 @@ func (l *List) ClearHighlight() {
 	l.lastHighlighted = make(map[int]bool)
 }
 
+// HandleKeyPress handles key press events for the currently selected item.
+// Returns true if the event was handled.
+func (l *List) HandleKeyPress(msg tea.KeyPressMsg) bool {
+	if l.selectedIdx < 0 || l.selectedIdx >= len(l.items) {
+		return false
+	}
+
+	if keyable, ok := l.items[l.selectedIdx].(KeyPressable); ok {
+		handled := keyable.HandleKeyPress(msg)
+		if handled {
+			l.invalidateItem(l.selectedIdx)
+		}
+		return handled
+	}
+
+	return false
+}
+
+// UpdateItems propagates a message to all items that implement Updatable.
+// This is typically used for animation messages like anim.StepMsg.
+// Returns commands from updated items.
+func (l *List) UpdateItems(msg tea.Msg) tea.Cmd {
+	var cmds []tea.Cmd
+	for i, item := range l.items {
```

Comment:
Iterating over all the items is expensive and resource consuming for large lists. We should only invalidate affected items _lazily_ as they come to view. Otherwise, we need to explore other options.